content/articles/how-to-build-a-vue-app-with-flask-sqlite-backend-using-docker/index.md

```diff
@@ -0,0 +1,952 @@
+
+*Add todo page*
+
+![add-todo-page](/engineering-education/how-to-build-a-vue-app-with-flask-sqlite-backend-using-docker/add-todos-page.png)
+
+### Dockerizing the application
+To dockerize the application we have built, we will follow the following steps:
+
+#### Dockerize the Flask API
+To dockerize the API, create a Dockerfile, and a `.dockerignore` file in the API folder.
+
+The Dockerfile will host the instructions for creating the image, whereas the `.dockerignore` file will host the files to be ignored when copying to the image.
+
+Add the following in the Dockerfile:
+
+```dockerfile
+# Base python package
+FROM python:3.8-slim-buster
+
+# Working directory
+WORKDIR /app
+
+# Copy the dependencies
+COPY requirements.txt requirements.txt
+
+# Install the dependencies
+RUN pip3 install -r requirements.txt
+
+# Copy the files
+COPY . .
+
+# Executable commands
+CMD [ "python3", "-m" , "flask", "run", "--host=0.0.0.0"]
+```
+
+From above, we are externally importing the python package, defining the working directory, copying the dependencies, installing the dependencies, copying the files, and setting the execution commands.
+
+Add the following to the `.dockerignore` file:
+
+```docker
+__pycache__/
+
+.gitignore
+
+Pipfile
+
+Pipfile.lock
+
+README.MD
+```
+
+From above, we are adding all the files that should not be included in the docker image.
+
+#### Dockerize the Vue app
+To dockerize the Vue app, we will also create a Dockerfile in the project folder. Similarly, as in the previous instance, it will host the instructions for creating the docker image.
+
+In the Dockerfile, add the following:
+
+```dockerfile
+#Base image
+FROM node:lts-alpine
+
+#Install serve package
+npm i -g serve
+
+# Set the working directory
+WORKDIR /app
+
+# Copy the package.json and package-lock.json
+COPY package*.json ./
+
+# install project dependencies
+RUN npm install
+
+# Copy the project files
+COPY . .
+
+# Build the project
+RUN npm run build
+
+# Expose a port
+EXPOSE 5000
+
+# Executables
+CMD [ "serve", "-s", "dist" ]
+```
+
+From above, we are importing the node image, setting the working directory, copying the contents of the `package.json`, and `package-lock.json`, Installing the project dependencies, copying the project files, building the project,  exposing a port, and setting the executables.
+
+#### Set up an overall docker-compose file
+After setting a Dockerfile for each of the folders i.e `api-folder`, and `client-folder`, we will set up a `docker-compose.yml` file outside the two folders.
+
+Start by creating a `docker-compose.yml` file outside the API and the client folder.
+
+In the `docker-compose.yml` file, add the following:
+
+```yml
+version: '3.8'
+
+services:        
+    flask-todos-api:
+        build: ./name_of_your_api_folder
+        ports: 
+            - 5000:5000
+
+    vue-todos-app:
+        build: ./name_of_your_client_folder
+        ports: 
```

Comment:
You used vue CLI to generate the Vue app. Replace the name you used earlier with `name_of_your_client_folder`